tests/test_read_configs_include.py

```diff
@@ -0,0 +1,265 @@
+import io
+import os
+import tempfile
+import pytest
+import subprocess
+
+ENV_SPLIT_STR: str = "#env for testing#"
+
+
+@pytest.fixture(scope="module")
+def root_dir(request):
+    return os.path.dirname(os.path.dirname(request.fspath))
+
+
+@pytest.fixture(scope="module")
+def read_config_include_file(root_dir) -> str:
+    return os.path.join(root_dir, "birdhouse", "read-configs.include.sh")
+
+
+def set_local_env(env_file: io.FileIO, content: str | dict) -> None:
+    env_file.truncate()
+    if isinstance(content, dict):
+        env_file.write("\n".join(f"{k}={v}" for k, v in content.items()))
+    else:
+        env_file.write(content)
+
+
+def split_and_strip(s: str, split_on="\n") -> list[str]:
+    return [cline for line in s.split(split_on) if (cline := line.strip())]
+
+
+def get_read_config_stdout(proc: subprocess.CompletedProcess) -> str:
+    return proc.stdout.split(ENV_SPLIT_STR)[0]
+
+
+def get_command_stdout(proc: subprocess.CompletedProcess) -> str:
+    return proc.stdout.split(ENV_SPLIT_STR)[1]
+
+
+class TestReadConfigs:
```

Comment:
Maybe only one test case missing for "other extras not within birdhouse".
They should be last, after components and optional-components, as final overrides.

Reply:
Sure I can do that but I'm not sure what that test adds? Anything in EXTRA_CONF_DIRS is treated the same regardless of whether its "in birdhouse" or a separate component. So it will be last if it's last in the list, there is no special treatment other than that.

Reply:
I added an additional test that just makes sure we can include components in some other arbitrary location